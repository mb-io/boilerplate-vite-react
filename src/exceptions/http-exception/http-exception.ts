import { AxiosError } from 'axios'

export class HttpException extends Error {
  status: number
  message: string
  error?: AxiosError

  constructor(status: number, message: string, error?: AxiosError) {
    super(message)
    this.status = status
    this.message = message
    this.error = error
  }
}
