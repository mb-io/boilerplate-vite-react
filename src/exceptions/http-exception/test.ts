import { describe, it, expect } from 'vitest'
import { HttpException } from './http-exception'
import { AxiosError } from 'axios'

describe('HttpException', () => {
  it('should create an exception with status and message', () => {
    const status = 400
    const message = 'Bad Request'
    const exception = new HttpException(status, message)
    expect(exception.status).toBe(status)
    expect(exception.message).toBe(message)
    expect(exception).toMatchSnapshot()
  })

  it('should create an exception with status, message, and AxiosError', () => {
    const status = 500
    const message = 'Internal Server Error'
    const axiosError = new AxiosError('Axios error message')
    const exception = new HttpException(status, message, axiosError)
    expect(exception.status).toBe(status)
    expect(exception.message).toBe(message)
    expect(exception.error).toBe(axiosError)
    expect(exception).toMatchSnapshot()
  })
})
