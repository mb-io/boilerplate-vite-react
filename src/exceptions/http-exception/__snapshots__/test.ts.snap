// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HttpException > should create an exception with status and message 1`] = `[Error: Bad Request]`;

exports[`HttpException > should create an exception with status, message, and AxiosError 1`] = `[Error: Internal Server Error]`;
